test/models/article_test.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ArticleTest < ActiveSupport::TestCase
+  def setup
+    @subscription_offer = articles(:one)
+  end
+
+  test 'should be valid' do
+    assert_predicate @subscription_offer, :valid?
+  end
+
+  test 'should not be valid without name' do
+    @subscription_offer.name = nil
+    assert_not_predicate @subscription_offer, :valid?
+  end
+
+  test 'should not be valid without price' do
+    @subscription_offer.price = nil
+    assert_not_predicate @subscription_offer, :valid?
+  end
+
+  test 'price should be integer' do
+    @subscription_offer.price = 10.56
+    assert_not_predicate @subscription_offer, :valid?
+  end
+
+  test 'price should be positive' do
+    @subscription_offer.price = -5
+    assert_not_predicate @subscription_offer, :valid?
+  end
+
+  test 'article should soft delete' do
+    assert_no_difference 'Article.count' do
+      @subscription_offer.soft_delete
+    end
```

Comment:
would it be a more thorough test if we checked that the deleted_at field is nil before the soft delete and not nil after? As we have some logic requiring the deleted_at to be not nil